docs/practical-examples/forms.md

```diff
@@ -0,0 +1,191 @@
+`FormBuilder` instance is a common enough pattern to grant showing it here._ 
+
+
+#### ✅ Encouraged: Abstract the markup into a ViewComponent
+
+In order to clean up your `FormBuilder` and be able to test the markup in isolation, you can extract the code inside the form builder method into a `ViewComponent`. This is very similar to what [Rails' internals](https://github.com/rails/rails/blob/d3f4db9e95a475822c02b29241a5d07cbcff6fd5/actionview/lib/action_view/helpers/form_helper.rb#L1151){:target="_blank"} do to create their default form helpers.
+
+Let's extract the code from the previous example into a `ViewComponent`
+
+```ruby
+class YourCustomFormBuilder < ActionView::Helpers::FormBuilder
+  # Note: (attribute, **options) is an oversimplification of the arguments passed into
+  # a form builder method and a ViewComponent. Take a look at the links above to understand
+  # how default form builder methods work, which arguments they receive, etc.
+  def text_area_with_label(attribute, **options)
+    @template.render TextAreaWithLabelComponent.new(attribute: attribute, **options)
+  end
+end
+```
+
+```ruby
+class TextAreaWithLabelComponent < ApplicationComponent
+  def initialize(attribute:, **options)
+    @attribute = attribute
+  end
+  attr_reader :attribute
+end
+```
+
+```erb
+<!-- app/components/text_area_with_label_component.html.erb -->
+<div class='maybe-some-wrapper-class'>
+  <%= label_tag attribute, class: 'your-label-css-classes' %>
+  <%= text_area_tag attribute, class: 'your-text-area-classes' %>
+</div>
+```
+
+```erb
+<!-- app/components/comment_form_component.html.erb -->
+<%= form_with(model: comment, builder: YourCustomFormBuilder) do |form| %>
+  <%= form.text_area_with_label :content %>
+<% end %>
+```
+
+#### Writing input elements in vanilla HTML
+
+Nothing is really stopping you from using regular `<input>` tags instead of Rails' helper methods. However, it is important for your own implementations to follow the conventions set by other form helpers. If you plan to do so, please read the [Understanding Parameter Naming Conventions](https://guides.rubyonrails.org/form_helpers.html#understanding-parameter-naming-conventions){:target="_blank"} section of the `form_helpers` Rails guide.
+
+## Gotchas and Known Issues
+
+### Using multiple `FormBuilder`s in your application
+In a typical Rails application, all views rendered by controllers that inherit from your `ApplicationController` will use the default `FormBuilder` provided by the framework. However, you can override the default `FormBuilder` used by your application wherever you think necessary. We previously saw that you can do so using the `builder` argument in `form_with` but you can also configure it at a [`controller` level](https://edgeapi.rubyonrails.org/classes/ActionController/FormBuilder.html){:target="_blank"}.
+
+This is currently *not supported* by `ViewComponent`. To overcome this, apply the correct form builder each time to the `form_with` method.
```

Comment:
What part of this is not supported by ViewComponent? Do you have an example of something that doesn't work? I have a custom form builder in my application, and it renders components similar to how you have shown above. I set the form builder in my `ApplicationController` and it works fine. To me this is just the way Rails works. I'm not understanding what setting a form builder has to do with ViewComponent at all.

Reply:
Thanks for raising this @davekaro. I used this a few months ago and it didn't work correctly for me but maybe things have changed since then. I removed this section for now.

I'll test this with the current version of ViewComponent to make sure it's actually an issue and update the documentation in a separate PR if necessary. I think I jumped the gun there 🙈. Sorry and thanks for pointing it out!